src/odemis/driver/test/smaract_test.py

```
@@ -816,5 +815,137 @@ def test_reference_cancel(self):
+        )
+
+    def test_reference_missing_deactive_pos_metadata(self):
+        """If pos_deactive_after_ref=True but FAV_POS_DEACTIVE is missing, it should log a warning but NOT crash."""
+        self.dev._pos_deactive_after_ref = True
+        self.dev._metadata.pop("FAV_POS_DEACTIVE")
+
+        # It should succeed without throwing an exception
+        f = self.dev.reference()
+        f.result(timeout=60)
+
+        for a, referenced in self.dev.referenced.value.items():
```

Ruff (0.15.20)

[warning] Loop control variable `a` not used within loop body

Rename unused `a` to `_a`

(B007)